wordpress-mobile/WordPress-Android
# Site creation rework: Retry the site-creation depending on the error

Additional PR for #6929 

This PR adds the "Try again" button.

Implementation details:
* The SiteCreationService now has to be able to start the process from an intermediate phase of the whole flow. The code is refactored to in a way that hopefully makes it more clear that there is only one sequence essentially but we enter it at different points.
* The message sent out by the Service now holds a _payload_ field that depending on the phase itself can be a different type of object
* The "FAILURE" phase has the _previous_ state object as a payload. Useful for retrying the operation.
* All _in-progress_ states hold the new site's remote ID as the payload

To test:
No specific steps. Need to manually make each step fail and see that the UI offers the "Try again" button and the via a debugger (or logcat) verify that the process is indeed resumed from that step.
